odl/space/fspace.py

```diff
@@ -154,199 +182,293 @@ class FunctionSetVector(Operator):
 
     """Representation of a `FunctionSet` element."""
 
-    def __init__(self, fset, fcall=None, fapply=None):
+    def __init__(self, fset, fcall):
         """Initialize a new instance.
 
         Parameters
         ----------
         fset : `FunctionSet`
             The set of functions this element lives in
-        fcall : `callable`, optional
+        fcall : `callable`
             The actual instruction for out-of-place evaluation.
             It must return an `FunctionSet.range` element or a
             `numpy.ndarray` of such (vectorized call).
-        fapply : `callable`, optional
-            The actual instruction for in-place evaluation.
-            Its first argument must be the
-            `FunctionSet.range` element
-            or the array of such (vectorization) to which the
-            result is written.
-
-        *At least one of the arguments fcall and fapply` must
-        be provided.*
         """
-        if not isinstance(fset, FunctionSet):
-            raise TypeError('function set {!r} not a `FunctionSet` '
-                            'instance.'.format(fset))
-
-        if fcall is None and fapply is None:
-            raise ValueError('call function and apply function cannot '
-                             'both be `None`.')
-
-        if fcall is not None and not callable(fcall):
-            raise TypeError('call function {!r} is not callable.'
-                            ''.format(fcall))
-
-        if fapply is not None and not callable(fapply):
-            raise TypeError('apply function {!r} is not callable.'
-                            ''.format(fapply))
-
         self._space = fset
-        self._call = fcall
-        self._apply = fapply
-
-        # Todo: allow users to specify linear
```

Comment:
I did not have that broad scope when I wrote this code. For me a `FunctionSpaceVector` is a function `f : R^n --> F`, and in that case linear functions are exactly those which can be written as `f(x) = v^H x` for some vector `v in R^n`. These may be more interesting when one has a general Hilbert space `H` as domain, but here I guess they're quite boring. I don't see a good reason to maintain the `linear` flag just for those.